test/Market.t.sol

```diff
@@ -108,11 +108,12 @@ contract Market_swap_rNori_mint_failure is UpgradeableMarket {
       block.timestamp,
       uint8(holdbackPercentage)
     );
-
+    uint256 numberOfNRTsToPurchase = 1 ether;
     uint256 ownerPrivateKey = 0xA11CE;
     owner = vm.addr(ownerPrivateKey);
-    checkoutTotal = _market.calculateCheckoutTotal(1 ether);
-    rNoriToMint = (1 ether * holdbackPercentage) / 100;
+    checkoutTotal = _market.calculateCheckoutTotal(numberOfNRTsToPurchase);
+    uint256 noriFeeAmount = _market.calculateNoriFee(numberOfNRTsToPurchase);
+    rNoriToMint = ((checkoutTotal - noriFeeAmount) * holdbackPercentage) / 100;
```

Comment:
Might be useful to add an external view function for this logic (I believe the market contract is also internally doing this calculation making it easy for this to become decoupled later-- some DRY code would prevent that)

Reply:
a function like this only makes sense on a removal-specific basis - it doesn't actually apply at the purchase level because individual removals that are used to fulfill a purchase could have different holdback percentages.  The only reason we use this line of logic in the test is because we know that all the removals have the same holdback percentage and we're generating a value to be able to expect against. 

I don't think we can make an external view function that actually makes sense / is wroth it on its own.